testsuite/shared/src/main/java/org/jboss/as/test/shared/logging/TestLogHandlerSetupTask.java

```diff
@@ -94,7 +94,33 @@ public void tearDown(ModelControllerClient managementClient) throws Exception {
 
         // remove file, note this needs to be done after the operations have been executed as we need to ensure that
         // no FD's are open. This can be an issue on Windows.
-        Files.deleteIfExists(logPath);
+        // small retry/backoff loop to handle lingering OS file locks.
```

Comment:
Sadly, this is not going to work. The main issue here is that the file is still in use, whereas the server process is still running. You are invoking `logHandlerSetup.tearDown(client)` before the container gets stopped, so delaying the deletion here is not going to take any effect.

We need to understand why in other tests, the file is deleted without issues, and in this test, it is not.

If we try to workaround it, there are chances where we could be putting problems under the carpet that in the future will arise again.

I understand this is a test issue, and it is not related to the fix itself, so if we do not find the root cause, we could assume the error by catching the exception in your test case and creating a follow up Jira with the intention of being fixed sooner than later.